### PR TITLE
Handle constant sized array allocations using no extension

### PR DESCRIPTION
**Pull request by victor-eds**
In LLVM, array allocations might have constant size:

```llvm
%array = alloca i32, i64 4, align 4
```

Represent this kind of allocations using `OpVariable + OpBitcast`.

Before this patch, the `SPV_INTEL_variable_length_array` extension was used.

### CODE REVIEW

File: test/simple.ll [hunk 1]
```diff
@@ -126,6 +126,13 @@ entry:
   ret void
 }
 
+; CHECK: "test_array_alloca"
+; Function Attrs: nounwind
+define dso_local void @test_array_alloca() #0 {
+  %arr = alloca i32, i64 4, align 4
+  ret void
+}
```
Comment (victor-eds) [inline test/simple.ll:134]: As no extension is used, this test would fail if an extension was needed.

**Comment by victor-eds** [comment]
@MrSidims can I get a review here?

File: test/simple.ll [hunk 2]
```diff
@@ -126,6 +126,13 @@ entry:
   ret void
 }
 
+; CHECK: "test_array_alloca"
+; Function Attrs: nounwind
+define dso_local void @test_array_alloca() #0 {
```
Comment (MrSidims) [inline test/simple.ll:131]: Probably it's worth to outline the test in a separate file with checks for generated SPIR-V and LLVM IR after the translation. Reason: LLVM IR changes after roundtrip (which is fine) and this should be exposed in the test.

**Review by MrSidims — APPROVED** [review]
Thanks for fixing this! Please consider a test change

File: test/simple.ll (hunk 2 above)
Comment (victor-eds) [reply test/simple.ll]: Created standalone test

**Comment by victor-eds** [comment]
Failing tests not because of lines I've even touched.

**Comment by MrSidims** [comment]
Restarting CI

**Pull request merged**